core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -349,12 +350,7 @@ object Validator extends SchemaValidator {
           )
 
           val v2 =
-            if (
-              t match {
-                case Some(t0) => t0._isOneOfInput
-                case _        => false
-              }
-            )
+            if (t.isDefined && t.get._isOneOfInput)
```

Comment:
`exists`

Reply:
Or did you avoid it to prevent the allocation under Scala 3?

Reply:
Sorry forgot to respond, yeah that was the reason 👍

Reply:
I saw your thumb up 😄

Reply:
It's quite sad we have to do that though 😭

Reply:
Actually, I think here we can change it. This is only executed for input objects which are extremely few

Reply:
Done 👍